vault/testing.go

```diff
@@ -335,11 +335,17 @@ func TestAddTestPlugin(t testing.T, c *Core, name, testFunc string) {
 	}
 
 	sum := hash.Sum(nil)
-	c.pluginCatalog.directory, err = filepath.EvalSymlinks(os.Args[0])
+
+	// Determine plugin directory path
+	fullPath, err := filepath.EvalSymlinks(os.Args[0])
 	if err != nil {
 		t.Fatal(err)
 	}
-	c.pluginCatalog.directory = filepath.Dir(c.pluginCatalog.directory)
+	directoryPath := filepath.Dir(fullPath)
+
+	// Set core's plugin directory and plugin catalog directory
+	c.pluginDirectory = directoryPath
```

Comment:
Why do we have both of these?

Reply:
`setupPluginCatalog()` uses `c.pluginDirectory`. If there is a unseal process during the test, the plugin loads from an empty directory and plugin mount fails:

"core: failed to create mount entry: path=mock-0/ error=no plugin found with name: mock-plugin"

Reply:
Gotcha. 👍